### PR TITLE
Fix SCE with ChainerX and normalize

### PR DESCRIPTION
**Pull request by hvy**
`F.softmax_cross_entropy` with ChainerX ndarrays used to return wrong results when called with default arguments (`reduce='mean'`, `normalize=True` and `ignore_label=-1`) with ignored labels in the target `t`. The issue was that `normalize` was not accounted for. This PR fixes this issue and addresses some TODOs.

### TODO
- [x] Fix `forward_chainerx`.
- [x] Add tests.

### CODE REVIEW

**Comment by hvy** [comment]
This PR is blocking https://github.com/chainer/chainer/pull/8250.

File: chainer/functions/loss/softmax_cross_entropy.py [hunk 1]
```diff
@@ -111,15 +109,22 @@ def _is_chainerx_supported(self, input_arrays):
         return True
 
     def forward_chainerx(self, inputs):
-        # TODO(niboshi): Current implementation is only intended to support
-        # MNIST example.
         x, t = inputs
```
Comment (hvy) [inline chainer/functions/loss/softmax_cross_entropy.py]: Note that `x.ndim == 2` because of https://github.com/chainer/chainer/pull/8301/files#diff-7f9fda8b243258115ef7bdac145fec8dR106-R107.

**Comment by takagi** [comment]
pfnCI, test this please.

**Comment by takagi** [comment]
pfnCI, test this please.

**Comment by takagi** [comment]
LGTM, waiting for CI finish.

**Comment by takagi** [comment]
Wait making the backport PR of this before #8307 will be merged to `v6`.
